crates/shared/src/event_handling.rs

```diff
@@ -133,6 +135,8 @@ where
         // in one transaction.
         let mut have_deleted_old_events = false;
         while let Some(events_chunk) = events.next().await {
+            // Early return on error is important here so that the second !have_deleted_old_events
```

Comment:
nanonit: I had a hard time finding the return statement (it's part of the ?). If it's so important maybe explitly return?

Reply:
Will change comment to mention the `?` but I'd like to keep it.